### ISSUE TITLE
USBToolBox: XHCI: waitforMatchingService failed or timed out

### ISSUE THREAD

**Issue by jridder** [USBToolBox/kext#7]
Hi,
What does this error exactly mean? Is it a timeout looking for a controller?

**Comment by dhinakg** [comment]
Hi, this isn't exactly an error.

UTB uses 3 methods (in order) to try to find the controller it is looking for:

1. Enumerate all children of the PCI device and see if any of them are a controller
2. Use `IOService::waitForMatchingService` to wait for a controller matching our parameters for up to 1 second
3. Register a notification for when a controller matching our parameters is registered

It is impossible to know whether `waitforMatchingService` errored or whether it just timed out (ie. fast booting computers that load USBToolBox before the USB kexts) without monitoring time elapsed (which is not worth the effort). Since it could have either failed or timed out, and both are good to know for support purposes, this is why this statement is not a debug statement. However, if `waitForMatchingService` times out, the notification handler will work to find the controller.

tl;dr: This is not a fatal error, just something useful to log

**Comment by jridder** [comment]
Thank you!

**Comment by WorldBoose** [comment]
Help me
